test/unit/content-src/components/SectionMenu.test.jsx

```diff
@@ -238,4 +223,40 @@ describe("<SectionMenu>", () => {
         });
       });
   });
+  describe("dispatch expand section if section is collapsed and adding top site", () => {
+    const dispatch = sinon.stub();
+    const expectedExpandData = {
+      id: DEFAULT_PROPS.id,
+      value: { collapsed: false },
+    };
+    const { options } = shallow(
+      <SectionMenu
+        {...DEFAULT_PROPS}
+        collapsed={true}
+        dispatch={dispatch}
+        extraOptions={["AddTopSite"]}
+      />
+    )
+      .find(ContextMenu)
+      .props();
+    afterEach(() => dispatch.reset());
+
+    assert.equal(options[0].id, "newtab-section-menu-add-topsite");
+    options
+      .filter(o => o.id === "newtab-section-menu-add-topsite")
+      .forEach(option => {
+        it(`should dispatch an action to expand the seciton`, () => {
+          option.onClick();
+
+          const [expandCallArgs] = dispatch.firstCall.args;
+          const { type, data } = expandCallArgs;
+          assert.ok(type === "UPDATE_SECTION_PREFS");
+          assert.deepEqual(data, expectedExpandData);
+        });
+        it(`should dispatch the add topsite action after`, () => {
+          option.onClick();
+          assert.ok(dispatch.thirdCall.calledWith(option.action));
```

Comment:
we can assert dispatch is called twice assert.calledTwice(dispatch); and check first and second call args similar to dispatch.secondCall.args in L219 above

Reply:
Yeah we can do something similar. Dispatch here will get called four times so I'll check each pair of two for the action and then the userEvent